apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/profile/ProfileThread.java

```diff
@@ -0,0 +1,296 @@
+                profilingSegmentSlot[slot] = null;
+
+                // setting stop running
+                currentSlotSegment.setSegmentIsRunning(false);
+                find = true;
+                break;
+            }
+        }
+
+        // decrease profile count
+        if (find) {
+            currentExecutionContext.getCurrentProfilingCount().addAndGet(-1);
+        }
+    }
+
+    /**
+     * shutdown profiling thread
+     */
+    public void shutdown() {
+        running = false;
+    }
+
+    /**
+     * start profiling
+     * @param executionContext
+     */
+    private void profiling(ProfileTaskExecutionContext executionContext) throws InterruptedException {
+
+        int maxSleepPeriod = executionContext.getTask().getThreadDumpPeriod();
+        int minDurationThreshold = executionContext.getTask().getMinDurationThreshold();
+
+        // run loop when current task still running
+        long currentLoopStartTime = -1;
+        while (checkCanKeepRunning(executionContext)) {
+            currentLoopStartTime = System.currentTimeMillis();
+
+            // each all slot
+            for (ProfilingSegmentContext slot : executionContext.getProfilingSegmentSlot()) {
+                if (slot == null) {
+                    continue;
+                }
+
+                // check is already start dump stack
+                if (slot.getStartDump()) {
+
+                    // dump stack
+                    if (!dumpSegment(slot)) {
+                        stopSegmentProfile(slot.getSegment());
+                        continue;
+                    }
+
+                } else {
+
+                    // check segment running time
+                    if (System.currentTimeMillis() - slot.getProfilingStartTime() > minDurationThreshold) {
+                        slot.setStartDump(true);
+                    }
+
+                }
+            }
+
+            // sleep to next period
+            // if out of period, sleep one period
+            long needToSleep = (currentLoopStartTime + maxSleepPeriod) - System.currentTimeMillis();
+            needToSleep = needToSleep > 0 ? needToSleep : maxSleepPeriod;
+            Thread.sleep(needToSleep);
+        }
+    }
+
+    /**
+     * dump segemnt thread stack
+     * @param segmentContext
+     * @return
+     */
+    private boolean dumpSegment(ProfilingSegmentContext segmentContext) {
+        // dump stack
+        if (!checkSegmentProfilingCanContinue(segmentContext)) {
+            return false;
+        }
+
+        return dumpThread(segmentContext);
+    }
+
+    /**
+     * dump thread stack, and push data to backend
+     * @param segmentContext
+     * @return still can dump
+     */
+    private boolean dumpThread(ProfilingSegmentContext segmentContext) {
+        long currentTime = System.currentTimeMillis();
+        // dump thread
+        final StackTraceElement[] stackTrace = segmentContext.getProfilingThread().getStackTrace();
```

Comment:
No matter how many times you check the thread#isAlive, you still could face the thread is not inactive or dead at here. Try/catch this, and remove all unnecessary check.